Json.More/JsonSerializerOptionsExtensions.cs

```diff
@@ -78,10 +76,10 @@ public static void Write<T>(this JsonSerializerOptions options, Utf8JsonWriter w
 	/// <param name="value">The value to serialize.</param>
 	/// <param name="inputType">The type to serialize.</param>
 	/// <returns>The value that was converted.</returns>
+	[RequiresDynamicCode("Calls JsonSerializer.Serialize. Make sure the options object contains all relevant JsonTypeInfos before suppressing this warning.")]
+	[RequiresUnreferencedCode("Calls JsonSerializer.Serialize. Make sure the options object contains all relevant JsonTypeInfos before suppressing this warning.")]
```

Comment:
Could be avoided if you called `options.GetTypeInfo()` before passing the result to the  JsonSerializer method accepting JsonTypeInfo

Reply:
`options.GetTypeInfo()` doesn't exist in .Net Standard 2.0.

Reply:
It should be available to .NET standard 2.0, however that would require referencing a newer version of the STJ NuGet package (>= 7.0.0 for that particular API). I can see why you might want to stay with 6.0.0 as the minimum supported version, but the APIs added in >= 7.0.0 make it more manageable to work in AOT-sensitive components.

Reply:
We tried updating to v8 and had issues.  I'm not sure we've tried v7.

Reply:
Unless you have customers that specifically require 6.0.0 I would strongly recommend trying to move the minimum version to 8.0.0. It has APIs specifically catering to AOT-ready libraries (e.g. the JsonSerializer.IsReflectionEnabledByDefault property) and it should eliminate the need to multi-target (you can stick with just one netstandard2.0 target)

Reply:
Working on it, it seems mostly good now.  Not sure what the errors were before.

There is a discrepancy in double serialization in netcoreapp3.1.  For example `-4.2` is serialized correctly in net6 and net8, but netcoreapp3.1 gives `-4.2000000000000002`.

Reply:
There weren't really issues, just some "danger use at your own risk" kinds of warnings. I staged a PR with it here but abandoned it: https://github.com/gregsdennis/json-everything/pull/630.

Reply:
Oh... There a warning on using STJ v8 with net core 3.1:

> System.Text.Encodings.Web 8.0.0 doesn't support $(TargetFramework) and has not been tested with it. Consider upgrading your TargetFramework to net6.0 or later. You may also set &lt;SuppressTfmSupportBuildWarnings&gt;true&lt;/SuppressTfmSupportBuildWarnings&gt; in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk.

Reply:
Yeah, the 3.1 `double` issue is the only one.  Everything else is working.  I'll start a PR.  Not sure why this was so complicated before.

Reply:
Great! Once you do that, you should also be able to remove most, if not all, of the NET8_OR_GREATER checks in the code.

Reply:
Given that STJ8 doesn't support 3.1 and everything else seems to work, we're dropping 3.1 support as well.  See #640